### PR TITLE
Fix test_router_bgp skips on XR

### PR DESCRIPTION
**Pull request by HelenHZhang**

### CODE REVIEW

**Comment by glennmatthews** [comment]
Please provide output from running this test script (should be passing in full on both XR and NX after this change) and resolve the merge conflicts reported - otherwise this looks good to me. Thanks.

**Comment by HelenHZhang** [comment]
## The output On XR:

```
rtpfe1@agent-lab21-pm:~/cisco-network-node-utils$ ruby tests/test_router_bgp.rb -v -- agent-lab21-xr.cisco.com:57777 root lab
/home/rtpfe1/.rvm/gems/ruby-2.2.1/gems/grpc-0.11.0/lib/grpc/generic/client_stub.rb:173: warning: circular argument reference - parent
Run options: -v -- --seed 45322

# Running:


Node under test:
  - name  - agent-lab21-xr
  - type  - R-IOSXRV9000-RP
TestRouterBgp#test_create_valid_asn = 74.46 s = .
TestRouterBgp#test_default_graceful_restart = 4.81 s = .
TestRouterBgp#test_get_neighbor_fib_down_accelerate_not_configured = 5.58 s = .
TestRouterBgp#test_get_confederation_peers_not_configured = 4.69 s = .
TestRouterBgp#test_maxas_limit_vrf = 8.66 s = .
TestRouterBgp#test_default_enforce_first_as = 5.08 s = .
TestRouterBgp#test_get_cluster_id_not_configured = 13.32 s = .
TestRouterBgp#test_get_confederation_id_not_configured = 4.29 s = .
TestRouterBgp#test_default_confederation_peers = 4.32 s = .
TestRouterBgp#test_enforce_first_as = 6.29 s = .
TestRouterBgp#test_default_cluster_id = 4.84 s = .
TestRouterBgp#test_default_neighbor_fib_down_accelerate = 4.80 s = .
TestRouterBgp#test_timer_bestpath_limit_always_vrf = 7.93 s = .
TestRouterBgp#test_confederation_id_default = 5.68 s = .
TestRouterBgp#test_default_route_distinguisher = 16.93 s = .
TestRouterBgp#test_default_suppress_fib_pending = 4.90 s = .
TestRouterBgp#test_destroy = 6.39 s = .
TestRouterBgp#test_timer_bgp_keepalive_hold = 14.74 s = .
TestRouterBgp#test_set_get_suppress_fib_pending = 15.84 s = .
TestRouterBgp#test_get_reconnect_interval_default = 2.86 s = .
TestRouterBgp#test_get_timer_bestpath_limit_default = 3.64 s = .
TestRouterBgp#test_bestpath_not_configured_default = 4.89 s = .
TestRouterBgp#test_create_vrfname_zero_length = 1.81 s = .
TestRouterBgp#test_get_router_id_not_configured = 4.44 s = .
TestRouterBgp#test_cluster_id_vrf = 6.91 s = .
TestRouterBgp#test_default_log_neighbor_changes = 5.41 s = .
TestRouterBgp#test_timer_bestpath_limit_always_default = 4.69 s = .
TestRouterBgp#test_get_asnum = 4.83 s = .
TestRouterBgp#test_default_confederation_id = 4.92 s = .
TestRouterBgp#test_bestpath_default = 21.13 s = .
TestRouterBgp#test_default_router_id = 4.53 s = .
TestRouterBgp#test_cluster_id_default = 7.50 s = .
TestRouterBgp#test_default_shutdown = 5.18 s = .
TestRouterBgp#test_default_timer_bgp_keepalive_hold_default = 10.59 s = .
TestRouterBgp#test_confederation_peers_default = 15.27 s = .
TestRouterBgp#test_create_valid = 20.28 s = .
TestRouterBgp#test_get_timer_bestpath_limit_always_not_configured = 4.82 s = .
TestRouterBgp#test_log_neighbor_changes_vrf = 8.58 s = .
TestRouterBgp#test_maxas_limit_default = 4.23 s = .
TestRouterBgp#test_collection_not_empty = 5.78 s = .
TestRouterBgp#test_set_get_shutdown = 5.01 s = .
TestRouterBgp#test_confederation_peers_vrf = 7.40 s = .
TestRouterBgp#test_create_vrf_invalid = 1.82 s = .
TestRouterBgp#test_bestpath_not_configured_vrf = 18.00 s = .
TestRouterBgp#test_graceful_restart_default = 11.70 s = .
TestRouterBgp#test_graceful_restart_vrf = 7.12 s = .
TestRouterBgp#test_confederation_id_vrf = 7.47 s = .
TestRouterBgp#test_set_get_reconnect_interval = 10.83 s = .
TestRouterBgp#test_set_get_timer_bestpath_limit = 9.89 s = .
TestRouterBgp#test_router_id = 12.47 s = .
TestRouterBgp#test_set_get_neighbor_fib_down_accelerate = 19.40 s = .
TestRouterBgp#test_get_suppress_fib_pending_not_configured = 4.39 s = .
TestRouterBgp#test_default_maxas_limit = 4.46 s = .
TestRouterBgp#test_create_invalid_multiple = 4.49 s = .
TestRouterBgp#test_bestpath_vrf = 15.09 s = .
TestRouterBgp#test_default_timer_bestpath_limit_always = 13.55 s = .
TestRouterBgp#test_log_neighbor_changes_default = 6.78 s = .
TestRouterBgp#test_route_distinguisher = 9.67 s = .
TestRouterBgp#test_collection_empty = 3.06 s = .
TestRouterBgp#test_set_get_confed_id_uu76828 = 5.11 s = .
TestRouterBgp#test_create_asnum_invalid = 1.93 s = .
TestRouterBgp#test_default_bestpath = 4.53 s = .
TestRouterBgp#test_get_shutdown_not_configured = 4.54 s = .

Finished in 554.553446s, 0.1136 runs/s, 0.3913 assertions/s.

63 runs, 217 assertions, 0 failures, 0 errors, 0 skips
```

## The output on Nexus:

```
rtpfe1@agent-lab21-pm:~/cisco-network-node-utils$ ruby tests/test_router_bgp.rb -v -- agent-lab5-nx.cisco.com admin admin
/home/rtpfe1/.rvm/gems/ruby-2.2.1/gems/grpc-0.11.0/lib/grpc/generic/client_stub.rb:173: warning: circular argument reference - parent
Run options: -v -- --seed 34374

# Running:


Node under test:
  - name  - agent-lab5-nx
  - type  - N9K-NXOSV
TestRouterBgp#test_create_invalid_multiple = 8.69 s = .
TestRouterBgp#test_set_get_reconnect_interval = 12.58 s = .
TestRouterBgp#test_confederation_peers_default = 14.75 s = .
TestRouterBgp#test_set_get_shutdown = 7.82 s = .
TestRouterBgp#test_get_timer_bestpath_limit_always_not_configured = 12.75 s = .
TestRouterBgp#test_set_get_confed_id_uu76828 = 8.46 s = .
TestRouterBgp#test_default_enforce_first_as = 7.62 s = .
TestRouterBgp#test_default_graceful_restart = 7.19 s = .
TestRouterBgp#test_destroy = 8.34 s = .
TestRouterBgp#test_create_vrfname_zero_length = 1.82 s = .
TestRouterBgp#test_graceful_restart_default = 15.02 s = .
TestRouterBgp#test_bestpath_vrf = 16.62 s = .
TestRouterBgp#test_maxas_limit_vrf = 13.28 s = .
TestRouterBgp#test_timer_bgp_keepalive_hold = 25.82 s = .
TestRouterBgp#test_confederation_id_default = 18.16 s = .
TestRouterBgp#test_default_log_neighbor_changes = 7.69 s = .
TestRouterBgp#test_get_router_id_not_configured = 12.69 s = .
TestRouterBgp#test_graceful_restart_vrf = 13.50 s = .
TestRouterBgp#test_collection_not_empty = 6.37 s = .
TestRouterBgp#test_default_cluster_id = 7.85 s = .
TestRouterBgp#test_create_valid = 12.68 s = .
TestRouterBgp#test_default_confederation_id = 7.56 s = .
TestRouterBgp#test_timer_bestpath_limit_always_vrf = 12.53 s = .
TestRouterBgp#test_default_timer_bgp_keepalive_hold_default = 7.92 s = .
TestRouterBgp#test_default_neighbor_fib_down_accelerate = 7.38 s = .
TestRouterBgp#test_timer_bestpath_limit_always_default = 8.13 s = .
TestRouterBgp#test_default_timer_bestpath_limit_always = 7.02 s = .
TestRouterBgp#test_create_asnum_invalid = 2.24 s = .
TestRouterBgp#test_enforce_first_as = 12.79 s = .
TestRouterBgp#test_get_neighbor_fib_down_accelerate_not_configured = 7.67 s = .
TestRouterBgp#test_get_suppress_fib_pending_not_configured = 12.08 s = .
TestRouterBgp#test_router_id = 13.12 s = .
TestRouterBgp#test_cluster_id_default = 7.68 s = .
TestRouterBgp#test_default_bestpath = 7.46 s = .
TestRouterBgp#test_get_asnum = 7.77 s = .
TestRouterBgp#test_get_cluster_id_not_configured = 8.13 s = .
TestRouterBgp#test_log_neighbor_changes_vrf = 13.13 s = .
TestRouterBgp#test_set_get_suppress_fib_pending = 13.56 s = .
TestRouterBgp#test_get_confederation_id_not_configured = 12.69 s = .
TestRouterBgp#test_default_router_id = 12.54 s = .
TestRouterBgp#test_route_distinguisher = 20.81 s = .
TestRouterBgp#test_bestpath_not_configured_default = 7.05 s = .
TestRouterBgp#test_default_confederation_peers = 7.67 s = .
TestRouterBgp#test_default_suppress_fib_pending = 7.76 s = .
TestRouterBgp#test_get_shutdown_not_configured = 8.10 s = .
TestRouterBgp#test_bestpath_not_configured_vrf = 7.06 s = .
TestRouterBgp#test_collection_empty = 2.30 s = .
TestRouterBgp#test_confederation_id_vrf = 7.71 s = .
TestRouterBgp#test_create_valid_asn = 84.07 s = .
TestRouterBgp#test_maxas_limit_default = 8.46 s = .
TestRouterBgp#test_get_reconnect_interval_default = 7.76 s = .
TestRouterBgp#test_bestpath_default = 11.44 s = .
TestRouterBgp#test_default_maxas_limit = 7.90 s = .
TestRouterBgp#test_create_vrf_invalid = 1.90 s = .
TestRouterBgp#test_get_timer_bestpath_limit_default = 7.57 s = .
TestRouterBgp#test_default_shutdown = 7.38 s = .
TestRouterBgp#test_log_neighbor_changes_default = 13.66 s = .
TestRouterBgp#test_get_confederation_peers_not_configured = 7.67 s = .
TestRouterBgp#test_set_get_timer_bestpath_limit = 14.02 s = .
TestRouterBgp#test_cluster_id_vrf = 8.70 s = .
TestRouterBgp#test_set_get_neighbor_fib_down_accelerate = 12.99 s = .
TestRouterBgp#test_confederation_peers_vrf = 9.23 s = .
TestRouterBgp#test_default_route_distinguisher = 7.66 s = .

Finished in 699.936397s, 0.0900 runs/s, 0.3457 assertions/s.

63 runs, 242 assertions, 0 failures, 0 errors, 0 skips
Warning: coverage data provided by Coverage [786] exceeds number of lines in /home/rtpfe1/cisco-network-node-utils/lib/cisco_node_utils/bgp.rb [778]
Coverage report generated for MiniTest to /home/rtpfe1/cisco-network-node-utils/coverage. 1228 / 1763 LOC (69.65%) covered.
Warning: coverage data provided by Coverage [786] exceeds number of lines in /home/rtpfe1/cisco-network-node-utils/lib/cisco_node_utils/bgp.rb [778]
Warning: coverage data provided by Coverage [786] exceeds number of lines in /home/rtpfe1/cisco-network-node-utils/lib/cisco_node_utils/bgp.rb [778]
```

**Comment by HelenHZhang** [comment]
@glennmatthews I fixed the merge conflict.

**Comment by glennmatthews** [comment]
:+1: Will merge tomorrow if no one else has any concerns before then. Thanks!

(BTW: you did re-run the test after fixing the merge conflict, right?)

**Comment by robert-w-gries** [comment]
@HelenHZhang Do you know what caused the merge conflict?  The commit log shows that "Fix test_router_bgp skips on XR" was committed twice, so I'm curious what the merge actually did.

**Comment by HelenHZhang** [comment]
I found a merge error I did not see it before. The attribute name "neighbor_fib_down_accelerate" was renamed as "neighbor_down_fib_accelerate". I re-commit it. Sorry.

File: tests/test_router_bgp.rb [hunk 1]
```diff
@@ -794,12 +800,20 @@ def test_set_get_neighbor_down_fib_accelerate
...
+      if platform == :ios_xr
+        assert_raises(Cisco::UnsupportedError) do
+          bgp.neighbor_down_fib_accelerate = true
+        end
+      else
+        bgp.neighbor_down_fib_accelerate = true
+        assert(bgp.neighbor_down_fib_accelerate,
+               "vrf #{@vrf}: bgp neighbor_down_fib_accelerate "\
+               'should be enabled')
+        bgp.neighbor_donw_fib_accelerate = false
```
Comment (glennmatthews) [inline tests/test_router_bgp.rb]: typo - s/donw/down

**Pull request merged**